thirdparty_builtin/CMakeLists.txt

```diff
@@ -181,3 +153,32 @@ if(ENABLE_TESTS)
     endif()
 
 endif()
+
+if(ENABLE_BENCHMARKS)
+    if(WIN32 AND BUILD_SHARED_LIBS)
+      message(FATAL_ERROR "Benchmarks cannot be built when BUILD_SHARED_LIBS=On")
+    endif()
+
+    ## google benchmark support
+    add_subdirectory(gbenchmark-master-2017-05-19
+                     ${BLT_BUILD_DIR}/thirdparty_builtin/gbenchmark-master-2017-05-19)
+
+    if (UNIX AND NOT APPLE)
+      find_library(RT_LIBRARIES rt)
+    endif()
+
+    blt_register_library(NAME gbenchmark
+                         INCLUDES ${benchmark_SOURCE_DIR}/include ${benchmark_SOURCE_DIR}
+                         LIBRARIES benchmark ${RT_LIBRARIES}
+                         TREAT_INCLUDES_AS_SYSTEM ON
+                         )
+
+    list(APPEND _blt_tpl_targets benchmark)
+
+    # This sets up a target to run the benchmarks
+    add_custom_target(run_benchmarks 
+                      COMMAND ctest -C Benchmark -VV -R benchmark
```

Comment:
Can ctest still be used if `enable_testing()` has not been called? @white238

Reply:
It is difficult to tell offhand -- @mcfadden8 can you please try this manually and see what happens?

To maintain this feature, perhaps we should add a configuration to travis that has benchmarks enabled and testing disabled?  (The downside is longer CI waiting times).

Reply:
The run_benchmarks target doesn't appear to work (even after adding the ENABLE_FOLDERS logic).  I see the following when ENABLE_TESTS=Off:

```script
$ make run_benchmarks
Scanning dependencies of target run_benchmarks
UpdateCTestConfiguration  from :/g/g0/martymcf/umpire/build/linux-rhel7-x86_64/umpire/DartConfiguration.tcl
UpdateCTestConfiguration  from :/g/g0/martymcf/umpire/build/linux-rhel7-x86_64/umpire/DartConfiguration.tcl
Test project /g/g0/martymcf/umpire/build/linux-rhel7-x86_64/umpire
Constructing a list of tests
Checking test dependency graph...
Checking test dependency graph end
No tests were found!!!
```